includes/class-wc-gateway-komoju-block.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+use Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType;
+
+final class WC_Gateway_Komoju_Blocks extends AbstractPaymentMethodType
+{
+	private $gateway;
+	protected $name;
+	protected $settings;
+
+	public function __construct(WC_Gateway_Komoju_Single_Slug $gateway)
+	{
+		$this->gateway = $gateway;
+		$this->name = $gateway->id;
+	}
+
+	public function initialize()
+	{
+		$this->settings = $this->gateway->payment_method['settings'] ?? [];
+		$this->settings = get_option('woocommerce_test_komoju_settings', []);
+	}
+
+	public function is_active()
+	{
+		return $this->gateway->is_available();
+	}
+
+	public function get_payment_method_script_handles()
+	{
+		wp_register_script(
+			'komoju-payment-blocks-integration',
+			plugin_dir_url(__FILE__) . './js/komoju-checkout-blocks.js',
+			[
+				'wc-blocks-registry',
+				'wc-settings',
+				'wp-element',
+				'wp-html-entities',
+			],
+			null,
+			true
+		);
+
+		return ['komoju-payment-blocks-integration'];
+	}
+
+	public function get_payment_method_data()
+	{
+		if (!is_checkout()) {
+			return;
+		}
+
+		// We lazily fetch one session to be shared by all payment methods with dynamic fields.
```

Comment:
This season will expire after 1 day. We should create at least one per order.

Reply:
I re-used this code form `komoju-single-slug.php` [Link](https://github.com/degica/komoju-woocommerce/pull/111/files/6c8d828616c9be6a42f0e372dae78a2984f78319#diff-687e98a5de34c807482e7df8eb7efa950c1592dd5cb360fac395ca433048929eR159)

I am creating one session per order, when the session is null. And this will be disposed with the new order.

```json
"session":"{\"id\":\"dtz7ht47hox48nn9efew4crxk\",...
```
Komoju-blocks are sharing same session, but when making a new order, I checked the session id changes
```json
"session":"{\"id\":\"evibmddzz7sphnbad9sgk7l7r\",\"
```

Since I am not a PHP master, I was also100% sure how php $static works, so I asked in the ChatGPT.

----
The $checkout_session variable in your PHP code is declared as a static variable within the get_payment_method_data() function. This means it behaves slightly differently than a regular local variable:

Persistence: A static variable in a function retains its value between function calls. This means that once $checkout_session is set (i.e., once the create_session_for_fields() method is called and returns a session), it will keep that value in subsequent calls to get_payment_method_data() during the lifecycle of the script.

Scope: The scope of the $checkout_session variable is limited to the get_payment_method_data() function, but its value persists across multiple executions of this function within the same request/script execution.

Lifetime: The lifetime of $checkout_session is tied to the script execution. It does not "last forever" but lasts as long as the script is running. Once the PHP process terminates (which typically happens at the end of a request or when a script finishes executing), the static variable is destroyed. Thus, $checkout_session does not persist across different requests or users.

Concurrency: Each request to a PHP script is generally handled independently (especially in common setups like with Apache or Nginx using PHP-FPM), meaning each request has its own instance of all variables, including static ones. Therefore, $checkout_session will be unique for each request and does not share state across different requests or users.

----

So, it seems fine.

Reply:
Ah! Right, PHP just executes the script once per request... Forgot about that. Thanks for the reminder. This looks fine indeed. Also I didn't realize it was already doing it this way - so definitely safe to keep the current behavior.